src/query/service/tests/it/storages/fuse/operations/optimize.rs

```diff
@@ -19,25 +19,38 @@ use crate::storages::fuse::table_test_fixture::append_sample_data;
 use crate::storages::fuse::table_test_fixture::append_sample_data_overwrite;
 use crate::storages::fuse::table_test_fixture::check_data_dir;
 use crate::storages::fuse::table_test_fixture::execute_command;
-use crate::storages::fuse::table_test_fixture::history_should_have_only_one_item;
+use crate::storages::fuse::table_test_fixture::history_should_have_item;
 use crate::storages::fuse::table_test_fixture::TestFixture;
 
 #[tokio::test]
 async fn test_fuse_snapshot_optimize() -> Result<()> {
-    do_purge_test("implicit pure", "").await
+    do_purge_test("implicit pure", "", 1, 0, 1, 1, 1).await
 }
 
 #[tokio::test]
 async fn test_fuse_snapshot_optimize_purge() -> Result<()> {
-    do_purge_test("explicit pure", "purge").await
+    do_purge_test("explicit pure", "purge", 1, 0, 1, 1, 1).await
+}
+
+#[tokio::test]
+async fn test_fuse_snapshot_optimize_statistic() -> Result<()> {
+    do_purge_test("explicit pure", "statistic", 3, 1, 2, 2, 2).await
```

Comment:
are **3** snapshots supposed to be left there, after one insertion and a `optimize stats` operation?

failed to figure out why the number of segments and blocks also increased, would you please add some docs here?

Reply:
Since `statistic`  operation:
1. generate a new snapshot
2. and does not purge the last snapshot

so,  after `statistic`, snapshot count will be add from 2 to 3, i add a test that after `statistic` operation do a purge, and snapshot will become to 1.